arangod/Indexes/IndexFactory.cpp

```diff
@@ -332,34 +344,27 @@ IndexId IndexFactory::validateSlice(velocypack::Slice info, bool generateKey,
     THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_BAD_PARAMETER,
                                    "expecting object for index definition");
   }
-
-  IndexId iid = IndexId::none();
-  auto value = info.get(StaticStrings::IndexId);
-
-  if (value.isString()) {
-    iid = IndexId{basics::StringUtils::uint64(value.copyString())};
-  } else if (value.isNumber()) {
-    iid = IndexId{basics::VelocyPackHelper::getNumericValue<IndexId::BaseType>(
-        info, StaticStrings::IndexId.c_str(), 0)};
-  } else if (!generateKey) {
+  auto const id = helpers::extractId(info);
+  if (id.isSet()) {
+    return id;
+  }
+  if (!generateKey) {
     // In the restore case it is forbidden to NOT have id
     THROW_ARANGO_EXCEPTION_MESSAGE(
         TRI_ERROR_BAD_PARAMETER,
         "cannot restore index without index identifier");
   }
-
-  if (iid.empty() && !isClusterConstructor) {
+  if (!isClusterConstructor) {
     // Restore is not allowed to generate an id
     VPackSlice type = info.get(StaticStrings::IndexType);
     // dont generate ids for indexes of type "primary"
     // id 0 is expected for primary indexes
```

Comment:
This comment was and is liar.

We return "none", uint64_max, not 0.
But I think this code never called for primary

Because in createIndex we have assert(type != primary)

Reply:
Well, this comment is true. It never tells that primary index id is generated here :)

Reply:
But it tells we should return 0? Not none?

Reply:
IMHO primary indexes will never be generated upon user request. They are already created before, and they will always get the id `0`.

Reply:
In such case needs to remove if and comment, if this code never called for primary index

Reply:
Buy probably better in separate pr